src/main/webapp/app/assessment/assessment-filters/assessment-filters.component.ts

```diff
@@ -48,4 +50,11 @@ export class AssessmentFiltersComponent {
     private static isSubmissionLocked(submission: Submission) {
         return submission && getLatestSubmissionResult(submission) && !getLatestSubmissionResult(submission)!.completionDate;
     }
+
+    ngOnChanges() {
+        this.filteredSubmissions = this.submissions.filter((submission) => {
+            const result = getLatestSubmissionResult(submission);
+            return result && !result.completionDate;
```

Comment:
Why aren't you using the `isSubmissionLocked` method here? It does the same as your code, so using it reduces code duplication.

Reply:
Also, does this have to be run on every change? This would also run this code whenever the filter changes (which would be redundant, as we already know that the result will be >1).
You should either move this into ngOnInit or first check if the submissions were changed as part of the OnChanges event. You can do this by adding and checking the `SimpleProps` to the `ngOnChanges` method: `ngOnChanges(changes: SimpleChanges)`